src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -0,0 +1,116 @@
+import h5py
+import numpy as np
+from suite2p.registration.register import pick_initial_reference
+from suite2p.registration.rigid import (apply_masks, compute_masks, phasecorr,
+                                        phasecorr_reference, shift_frame)
+
+
+def load_initial_frames(file_path: str,
+                        h5py_key: str,
+                        nimg_init: int) -> np.ndarray:
+    """Load a subset of frames from the hdf5 data specified by file_path.
+
+    Parameters
+    ----------
+    file_path : str
+        Location of the raw 2Photon, HDF5 data to load.
+    h5py_key : str
+        Name of the dataset to load from the HDF5 file.
+    nimg_init : int
+        Number of frames to load from the input HDF5 data.
+
+    Returns
+    -------
+    frames : array-like, (nimg_init, Ly, Lx)
+        Selected frames from the input raw data linearly spaced in index of the
+        time axis. If nimg_init > nframes, a number of frames equal to nframes
+        is returned.
+    """
+    with h5py.File(file_path, "r") as hdf5_file:
+        # Get a set of linear spaced frames to load from disk.
+        tot_frames = hdf5_file[h5py_key].shape[0]
+        requested_frames = np.linspace(0,
+                                       tot_frames,
+                                       1 + min(nimg_init, tot_frames),
+                                       dtype=int)[:-1]
+        frames = hdf5_file[h5py_key][requested_frames]
+    return frames
+
+
+def compute_reference(frames: np.ndarray,
+                      niter: int,
+                      maxregshift: float,
+                      smooth_sigma: float,
+                      smooth_sigma_time: float) -> np.ndarray:
+    """Computes a stacked reference image from the input frames.
+
+    Modified version of Suite2P's compute_reference function with no updating
+    of input frames. Picks initial reference then iteratively aligns frames to
+    create reference. Does not reproduce the Suite2p 1Photon code path.
+
+    Parameters
+    ----------
+    frames : array-like, (nimg_init, Ly, Lx)
+        Set of frames to create a reference from.
+    niter : int
+        Number of iterations to perform when creating the reference image.
+    maxregshift : float
+        Maximum shift allowed as a fraction of the image width or height, which
+        ever is longer.
+    smooth_sigma : float
+        Width of the Gaussian used to smooth the phase correlation between the
+        reference and the frame with which it is being registered.
+    smooth_sigma_time : float
+        Width of the Gaussian used to smooth between multiple frames by before
+        phase correlation.
+
+    Returns
+    -------
+    refImg : array-like, (Ly, Lx)
+        Reference image created from the input data.
+    """
+    # Get initial reference image from suite2p.
+    ref_image = pick_initial_reference(frames)
+
+    for idx in range(niter):
```

Comment:
Is it always possible to know ahead of time what a good value of `niter` should be? Is it possible to check for convergence by checking if the reference image doesn't change that much between iterations?

Reply:
That's the plan going forward in #394. For now, the 8 default from suite2p seems to be enough in most cases. I played around a bit with convergence stopping criteria and 8 seems to be around the right number. I've never seen it take longer than 20 for a reasonable convergence criteria.